runtime/src/accounts_index.rs

```diff
@@ -1194,12 +1195,21 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
                     max_clean_root,
                     account_indexes,
                 );
+                empty_key = slot_list.is_empty();
             });
         }
+        if empty_key {
+            let mut w_maps = self.account_maps.write().unwrap();
+            if let Some(x) = w_maps.get(pubkey) {
+                if x.slot_list.read().unwrap().is_empty() {
+                    w_maps.remove(pubkey);
+                }
+            }
+        }
     }
 
-    pub fn can_purge(max_root: Slot, slot: Slot) -> bool {
-        slot < max_root
+    pub fn can_purge(max_clean_root: Slot, max_present_root: Slot, slot: Slot) -> bool {
```

Comment:
nits: I think we can remove the `pub` and also rename `can_purge()` to `can_purge_older_entries()` since it's only specifically used by `purge_older_root_entries()`

Reply:
Also nit: rename `max_present_root` to `max_cleanable_root_in_slot_list`

Reply:
For the `can_purge()` function, there's not a slot list, so should the rename still be `max_cleanable_root_in_slot_list`? I also don't think it should be `max_cleanable_root`, since the other parameter is called `max_clean_root` and that may be confusing (it is to me, at least!).

Reply:
OK, this was changed to `newest_root_in_slot_list`, as per @ryoqun 's suggestion below.